src/test/java/com/codeus/winter/config/PackageBeanRegistrationTest.java

```diff
@@ -1,95 +1,80 @@
-        registry = mock(BeanDefinitionRegistry.class);
-        beanRegistration = new PackageBeanRegistration(packageScanner, registry);
+        registry = new BeanDefinitionRegistryImpl();
+        beanRegistration = new PackageBeanRegistration(registry);
     }
 
-    /**
-     * Tests that classes annotated with {@link Component} are successfully
-     * registered as {@link BeanDefinition}s in the registry.
-     */
     @Test
     void shouldRegisterBeansSuccessfully() {
-        Class<?> testClass = WinterComponent.class;
-        when(packageScanner.findClassesWithAnnotations(anyString(), anySet()))
-                .thenReturn(Set.of(testClass));
-        when(registry.containsBeanDefinition("winterComponent")).thenReturn(false);
-
-        beanRegistration.registerBeans("com.framework");
+        beanRegistration.registerBeans("com.codeus.winter.config.test.inner");
 
-        verify(registry).registerBeanDefinition(eq("winterComponent"), any(BeanDefinitionImpl.class));
+        BeanDefinition definition = registry.getBeanDefinition("winterComponent");
+        assertNotNull(definition);
+        // bean name
+        assertEquals("com.codeus.winter.config.test.inner.WinterComponent", definition.getBeanClassName());
+        // scope
+        assertEquals(SCOPE_PROTOTYPE, definition.getScope());
+        assertFalse(definition.isSingleton());
+        // depends on
+        String[] dependencies = {"com.codeus.winter.config.test.inner.AutowiredComponent",
+            "com.codeus.winter.config.test.inner.QualifierComponent"
+        };
+        assertArrayEquals(dependencies, definition.getDependsOn());
+        // init method name
+        assertEquals("init", definition.getInitMethodName());
+        // destroy method name
+        assertEquals("destroy", definition.getDestroyMethodName());
     }
 
-    /**
-     * Tests that a {@link NotUniqueBeanDefinitionException} is thrown if two
-     * classes with the same bean name are registered.
-     */
     @Test
-    void shouldThrowExceptionForDuplicateBeanName() {
-        Class<?> testClass = WinterComponent.class;
-        when(packageScanner.findClassesWithAnnotations(anyString(), anySet()))
-                .thenReturn(Set.of(testClass));
-        when(registry.containsBeanDefinition("winterComponent")).thenReturn(true);
+    void shouldThrowExceptionForDuplicateBeanNamePassedSamePackageTwoTimes() {
+        beanRegistration.registerBeans("com.codeus.winter.config.test.inner");
+        assertThrows(NotUniqueBeanDefinitionException.class, () ->
+            beanRegistration.registerBeans("com.codeus.winter.config.test.inner")
+        );
+    }
 
+    @Test
+    void shouldThrowExceptionForDuplicateBeanNameFoundSameClassInDifferentPackages() {
         assertThrows(NotUniqueBeanDefinitionException.class, () ->
-                beanRegistration.registerBeans("com.framework")
+            beanRegistration.registerBeans("com.codeus.winter")
```

Comment:
I believe you meant to set `com.codeus.winter.config.test` because we don't want to scan all test packages - `com.codeus.winter`.

Reply:
fixed